test/e2e_test.go

```diff
@@ -0,0 +1,230 @@
+		})
+
+		By("Deploying Decco in the test cluster", func() {
+			out, err := execCommand("kubectl", "apply", "--namespace", "decco", "-f", testStagingManifestsDir)
+			Expect(err).To(BeNil(), "Failed to deploy Decco: %s", string(out))
+		})
+	})
+
+	AfterSuite(func() {
+		if os.Getenv("DECCO_TEST_PRESERVE_CLUSTER") == "" {
+			By("Cleaning up the test cluster (to preserve test cluster set DECCO_TEST_PRESERVE_CLUSTER to a non-empty value)")
+			out, err := execCommand("kind", "delete", "cluster", "--name", testClusterName, "--kubeconfig", testClusterKubeconfigPath)
+			Expect(err).To(BeNil(), "Failed to delete test KinD cluster: %s", string(out))
+		} else {
+			By("Preserving test KinD cluster: " + testClusterName)
+		}
+
+		if os.Getenv("DECCO_TEST_PRESERVE_DATA") == "" && os.Getenv("DECCO_TEST_PRESERVE_CLUSTER") == "" {
+			By("Cleaning up temporary test data (to preserve test data set DECCO_TEST_PRESERVE_DATA or DECCO_TEST_PRESERVE_CLUSTER to a non-empty value)")
+			err := os.RemoveAll(testStagingDir)
+			Expect(err).To(BeNil())
+		} else {
+			By("Preserving test data: " + testStagingDir)
+		}
+
+	})
+
+	Context("Decco deployment", func() {
+
+		It("should have all pods in a 'Running' state", func() {
+			Eventually(func() error {
+				out, err := execCommand("kubectl", "--namespace", "decco", "get", "pods", "--no-headers=true")
+				if err != nil {
+					wrappedErr := fmt.Errorf("failed to get pods: %s", string(out))
+					log(wrappedErr.Error())
+					return wrappedErr
+				}
+				lines := strings.Split(string(out), "\n")
+				if len(lines) == 0 {
+					return fmt.Errorf("no pods found")
+				}
+
+				for i := 0; i < len(lines); i++ {
+					line := strings.TrimSpace(lines[i])
+					if len(line) > 0 && !strings.Contains(line, "Running") {
+						wrappedErr := fmt.Errorf("pod is not in runnning state: '%s'", line)
+						log(wrappedErr.Error())
+						return wrappedErr
+					}
+				}
+				return nil
+			}, 5*time.Minute, 10*time.Second).Should(BeNil())
+		})
+
+		It("should create the Space CRD", func() {
+			Eventually(func() error {
+				out, err := execCommand("kubectl", "get", "crds", "--no-headers=true")
+				if err != nil {
+					wrappedErr := fmt.Errorf("failed to get crds: %s", string(out))
+					log(wrappedErr.Error())
+					return wrappedErr
+				}
+
+				for _, line := range strings.Split(string(out), "\n") {
+					if strings.Contains(line, "spaces.decco.platform9.com") {
+						return nil
+					}
+				}
+				return fmt.Errorf("space CRD not found")
+
+			}, 5*time.Minute, 10*time.Second).Should(BeNil())
+		})
+	})
+})
+
+func ensureCommandExists(name string) {
+	By("Checking if prerequisite command exists: " + name)
+	_, err := execCommand("which", name)
+	Expect(err).To(BeNil(), "Could not find prerequisite command '%s'", name)
+}
+
+func stageManifest(manifestPath string) {
+	By("Staging manifest: " + manifestPath)
+	out, err := execCommand("cp", manifestPath, testStagingManifestsDir)
```

Comment:
same as above. You can keep these files pre-created in testdata as well :). That trick is very handy

Reply:
There are some resources that I pull directly from `../manifests`. I copy them over to a temporary "test staging" directory to get an intermediate debuggable view of which exact resources will get installed in the test cluster (and to do so with a single `kubectl apply -f ${testManifestsDir}`. So it is a bit verbose, but that is intentional. ;)